### ISSUE TITLE
docker demo run: empty output dir

### ISSUE THREAD

**Issue by lgeistlinger** [waldronlab/curatedmetagenomics#32]
I followed the [instructions for running the docker container](https://github.com/waldronlab/curatedmetagenomics/wiki/Environment-variables-and-invocation) and executed:

```
docker run -it --env-file cmd_env_vars -v ${DB_PATH}:/biobakery_db -v ${OUTPUT_PATH}:/output waldronlab/curatedmetagenomics cmd_pipeline --demo TEST_SAMPLE ERR262957 /output /biobakery_db
```

which seems to complete without issues, reporting the creation of specific output files, for example:

```
Computing gene families ...

Computing pathways abundance and coverage ...

Output files created: 
/nobackup/16tb_b/curatedmetagenomics/output/humann/DEMO_genefamilies.tsv
/nobackup/16tb_b/curatedmetagenomics/output/humann/DEMO_pathabundance.tsv
/nobackup/16tb_b/curatedmetagenomics/output/humann/DEMO_pathcoverage.tsv
```

However, inspecting the chosen output dir, I find it to be empty and not containing these files:

```
lgeistlinger@wallabe4:~$ ls -a /nobackup/16tb_b/curatedmetagenomics/output
.  ..
```

**Comment by fbeghini** [comment]
`Output files created: [...]` is the default HUMAnN output, the output files are stored under `genefamilies/` `pathabundance/` `pathcoverage/`

**Comment by lgeistlinger** [comment]
Problem is `/nobackup/16tb_b/curatedmetagenomics/output/` (corresponding to my `OUTPUT_PATH`) is empty after the run, ie it doesn't contain the a `humann` subdir, nor the three subdirs within that you mention.

**Comment by fbeghini** [comment]
Sorry, I've missed the last part. I'll have a look. Which version is `cmd_pipeline` ?

**Comment by lgeistlinger** [comment]
Not sure how to obtain that. This is how my cmd_env_vars file looks:

```
DB_PATH=/nobackup/16tb_b/curatedmetagenomics/biobakery.db
OUTPUT_PATH=/nobackup/16tb_b/curatedmetagenomics/output
ncores=10  
chocophlanname=full_chocophlan.v296_201901.tar.gz
chocophlanurl=https://www.dropbox.com/s/das8hdof0zyuyh8/${chocophlanname}?dl=1
unirefname=uniref90_annotated_v201901.tar.gz
unirefurl=https://www.dropbox.com/s/yeur7nm7ej7spga/${unirefname}?dl=1
```

**Comment by fbeghini** [comment]
Something weird is happening with the volume binding: if you check the content inside the docker of the db folder, you have files
```
root@de04619d9d9c:/# ls /biobakery_db/
humann  metaphlan_db
```

on the host machine:
```
(test) [ 18:19:25 ] francesco.beghini@sci-cm4 (/shares/CIBIO-Storage/CM/news/tmp/francesco.beghini)
$ ls biobakery.db/
(test) [ 18:19:52 ] francesco.beghini@sci-cm4 (/shares/CIBIO-Storage/CM/news/tmp/francesco.beghini)
$
```

**Comment by seandavi** [comment]
The --env-file only affects the environment inside the docker container. If you haven't set DB_PATH and OUTPUT_PATH before running docker (that is, in your host shell), those variables are "unset" when you run the docker command. So:
1. Set the DB_PATH and OUTPUT_PATH in your HOST environment.
2. The env-file variables are set *inside* the container, not outside. Therefore, I suspect that setting the DB_PATH and OUTPUT_PATH to a host directory is not what you want.

**Comment by fbeghini** [comment]
DB_PATH and OUTPUT_PATH are set for me, but it seems that I have not specified the full path to the folder (`-v biobakery.db:/biobakery_db` instead of `-v /shares/CIBIO-Storage/CM/news/tmp/francesco.beghini/biobakery.db:/biobakery_db`)

**Comment by lgeistlinger** [comment]
@seandavi : well I ran

```
export $(cat cmd_env_vars | xargs)
```

as instructed in the host shell before calling docker run ... I can convince myself by typing:

```
lgeistlinger@wallabe4:~$ echo $DB_PATH
/nobackup/16tb_b/curatedmetagenomics/biobakery.db
lgeistlinger@wallabe4:~$ echo $OUTPUT_PATH
/nobackup/16tb_b/curatedmetagenomics/output
```

**Comment by lgeistlinger** [comment]
Maybe I indeed have to set these variables already before pulling

```
docker pull waldronlab/curatedmetagenomics
```
?

**Comment by seandavi** [comment]
That looks right to me, @lgeistlinger. Next step is to run the container using:

```
docker run -it --env-file cmd_env_vars -v ${DB_PATH}:/biobakery_db -v ${OUTPUT_PATH}:/output waldronlab/curatedmetagenomics /bin/bash
```

Now, check to see that the directories are doing what you expect. For example, try touching a file in /output to make sure that it ends up in the right place on the host system. 

Check the environment variables inside the container to make sure you got what you expect. 

Then run the cmd_pipeline script as usual inside the container. When the script completes, the docker container will still be running, so you can find the output files and see what the problem is.

**Comment by seandavi** [comment]
> Maybe I indeed have to set these variables already before pulling
> 
> ```
> docker pull waldronlab/curatedmetagenomics
> ```
> 
> ?

No, you don't need to do that. The environment is set up at docker run time, not at the time of the pull.

**Comment by lgeistlinger** [comment]
> No, you don't need to do that. The environment is set up at docker run time, not at the time of the pull.

Alright

**Comment by fbeghini** [comment]
Found the error, `output_path` is still taken from the env_file and not from the argument. I'm fixing this now

**Comment by fbeghini** [comment]
@lgeistlinger Can you try again with the latest Docker build (5e8c66c) ?

**Comment by lgeistlinger** [comment]
Do I have to modify my pull command for that?
Currently it says:
```
lgeistlinger@wallabe4:/nobackup/16tb_b/curatedmetagenomics$ docker pull waldronlab/curatedmetagenomics
Using default tag: latest
latest: Pulling from waldronlab/curatedmetagenomics
Digest: sha256:46ee1a6357c6b634b7c0715cd67c7d9f6a95e77b04102cb11eebd509a8d070dd
Status: Image is up to date for waldronlab/curatedmetagenomics:latest
docker.io/waldronlab/curatedmetagenomics:latest
```
which seems to indicate that there is nothing to update here ...

**Comment by seandavi** [comment]
`latest` doesn't do what you expect. `latest`, the default for `docker pull` means to use the latest version that worked. In this case, you have a cached version already available, so docker will use that. You'll need to drop in the docker hash or remove the locally cached image. 

```
docker pull waldronlab/curatedmetagenomics:46ee1a6357c6
```

Alternatively, 

```
docker rm -v waldronlab/curatedmetagenomics #remove local cache first
docker pull waldronlab/curatedmetagenomics
```

https://medium.com/@mccode/the-misunderstood-docker-tag-latest-af3babfd6375

**Comment by lgeistlinger** [comment]
Hm, I tried both routes, but I am running into errors:

```
lgeistlinger@wallabe4:/nobackup/16tb_b/curatedmetagenomics$ docker pull waldronlab/curatedmetagenomics:46ee1a6357c6
Error response from daemon: manifest for waldronlab/curatedmetagenomics:46ee1a6357c6 not found: manifest unknown: manifest unknown
```

```
lgeistlinger@wallabe4:/nobackup/16tb_b/curatedmetagenomics$ docker rm -v waldronlab/curatedmetagenomics
Error: No such container: waldronlab/curatedmetagenomics
```
